### ISSUE TITLE
Failing tests

### ISSUE THREAD

**Issue by adarshp** [clulab/tomcat-text#1]
Hi @pelovett ,

When I run `sbt test` on my computer, I get the following errors:

```
[info] TestParse:
[info] AsistEngine
[info] - should Parse close events properly *** FAILED ***
[info]   java.lang.IndexOutOfBoundsException: 0
[info]   at scala.collection.immutable.Vector.checkRangeConvert(Vector.scala:123)
[info]   at scala.collection.immutable.Vector.apply(Vector.scala:114)
[info]   at org.clulab.asist.TestParse.$anonfun$new$1(TestParse.scala:18)
[info]   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[info]   at org.scalatest.FlatSpecLike$$anon$1.apply(FlatSpecLike.scala:1682)
[info]   at org.scalatest.TestSuite.withFixture(TestSuite.scala:196)
[info]   ...
[info] - should Parse craft events properly *** FAILED ***
[info]   java.lang.IndexOutOfBoundsException: 0
[info]   at scala.collection.immutable.Vector.checkRangeConvert(Vector.scala:123)
[info]   at scala.collection.immutable.Vector.apply(Vector.scala:114)
[info]   at org.clulab.asist.TestParse.$anonfun$new$2(TestParse.scala:26)
[info]   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[info]   at org.scalatest.FlatSpecLike$$anon$1.apply(FlatSpecLike.scala:1682)
[info]   at org.scalatest.TestSuite.withFixture(TestSuite.scala:196)
[info]   ...
[info] - should Parse sight events properly *** FAILED ***
[info]   java.lang.IndexOutOfBoundsException: 0
[info]   at scala.collection.immutable.Vector.checkRangeConvert(Vector.scala:123)
[info]   at scala.collection.immutable.Vector.apply(Vector.scala:114)
[info]   at org.clulab.asist.TestParse.$anonfun$new$3(TestParse.scala:32)
[info]   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[info]   at org.scalatest.FlatSpecLike$$anon$1.apply(FlatSpecLike.scala:1682)
[info]   at org.scalatest.TestSuite.withFixture(TestSuite.scala:196)
[info]   ...
[info] - should Parse extinguish events properly *** FAILED ***
[info]   java.lang.IndexOutOfBoundsException: 0
[info]   at scala.collection.immutable.Vector.checkRangeConvert(Vector.scala:123)
[info]   at scala.collection.immutable.Vector.apply(Vector.scala:114)
[info]   at org.clulab.asist.TestParse.$anonfun$new$4(TestParse.scala:38)
[info]   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[info]   at org.scalatest.FlatSpecLike$$anon$1.apply(FlatSpecLike.scala:1682)
[info]   at org.scalatest.TestSuite.withFixture(TestSuite.scala:196)
[info]   ...
[info] - should Parse toggle events properly *** FAILED ***
[info]   java.lang.IndexOutOfBoundsException: 0
[info]   at scala.collection.immutable.Vector.checkRangeConvert(Vector.scala:123)
[info]   at scala.collection.immutable.Vector.apply(Vector.scala:114)
[info]   at org.clulab.asist.TestParse.$anonfun$new$5(TestParse.scala:45)
[info]   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[info]   at org.scalatest.FlatSpecLike$$anon$1.apply(FlatSpecLike.scala:1682)
[info]   at org.scalatest.TestSuite.withFixture(TestSuite.scala:196)
[info]   ...
[info] - should Parse save events properly *** FAILED ***
[info]   java.lang.reflect.InvocationTargetException:
[info]   at sun.reflect.GeneratedMethodAccessor28.invoke(Unknown Source)
[info]   at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[info]   at java.lang.reflect.Method.invoke(Method.java:498)
[info]   at scala.reflect.runtime.JavaMirrors$JavaMirror$JavaVanillaMethodMirror2.jinvokeraw(JavaMirrors.scala:384)
[info]   at scala.reflect.runtime.JavaMirrors$JavaMirror$JavaMethodMirror.jinvoke(JavaMirrors.scala:340)
[info]   at scala.reflect.runtime.JavaMirrors$JavaMirror$JavaVanillaMethodMirror.apply(JavaMirrors.scala:356)
[info]   at org.clulab.odin.impl.ActionMirror.$anonfun$reflect$1(ActionMirror.scala:23)
[info]   at org.clulab.odin.impl.GraphExtractor.findAllIn(Extractor.scala:84)
[info]   at org.clulab.odin.impl.Extractor.$anonfun$findAllIn$1(Extractor.scala:19)
[info]   at org.clulab.odin.impl.Extractor.$anonfun$findAllIn$1$adapted(Extractor.scala:18)
[info]   ...
[info]   Cause: java.lang.IndexOutOfBoundsException: 0
[info]   at scala.collection.mutable.ResizableArray.apply(ResizableArray.scala:43)
[info]   at scala.collection.mutable.ResizableArray.apply$(ResizableArray.scala:42)
[info]   at scala.collection.mutable.ArrayBuffer.apply(ArrayBuffer.scala:48)
[info]   at org.clulab.asist.StubActions.$anonfun$removeResearcher$1(AsistActions.scala:27)
[info]   at org.clulab.asist.StubActions.$anonfun$removeResearcher$1$adapted(AsistActions.scala:21)
[info]   at scala.collection.immutable.List.foreach(List.scala:389)
[info]   at org.clulab.asist.StubActions.removeResearcher(AsistActions.scala:21)
[info]   at sun.reflect.GeneratedMethodAccessor28.invoke(Unknown Source)
[info]   at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[info]   at java.lang.reflect.Method.invoke(Method.java:498)
[info]   ...
[info] - should Parse defeat events properly *** FAILED ***
[info]   java.lang.IndexOutOfBoundsException: 0
[info]   at scala.collection.immutable.Vector.checkRangeConvert(Vector.scala:123)
[info]   at scala.collection.immutable.Vector.apply(Vector.scala:114)
[info]   at org.clulab.asist.TestParse.$anonfun$new$7(TestParse.scala:60)
[info]   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[info]   at org.scalatest.FlatSpecLike$$anon$1.apply(FlatSpecLike.scala:1682)
[info]   at org.scalatest.TestSuite.withFixture(TestSuite.scala:196)
[info]   ...
[info] - should Parse search events properly *** FAILED ***
[info]   java.lang.reflect.InvocationTargetException:
[info]   at sun.reflect.GeneratedMethodAccessor28.invoke(Unknown Source)
[info]   at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[info]   at java.lang.reflect.Method.invoke(Method.java:498)
[info]   at scala.reflect.runtime.JavaMirrors$JavaMirror$JavaVanillaMethodMirror2.jinvokeraw(JavaMirrors.scala:384)
[info]   at scala.reflect.runtime.JavaMirrors$JavaMirror$JavaMethodMirror.jinvoke(JavaMirrors.scala:340)
[info]   at scala.reflect.runtime.JavaMirrors$JavaMirror$JavaVanillaMethodMirror.apply(JavaMirrors.scala:356)
[info]   at org.clulab.odin.impl.ActionMirror.$anonfun$reflect$1(ActionMirror.scala:23)
[info]   at org.clulab.odin.impl.GraphExtractor.findAllIn(Extractor.scala:84)
[info]   at org.clulab.odin.impl.Extractor.$anonfun$findAllIn$1(Extractor.scala:19)
[info]   at org.clulab.odin.impl.Extractor.$anonfun$findAllIn$1$adapted(Extractor.scala:18)
[info]   ...
[info]   Cause: java.lang.IndexOutOfBoundsException: 0
[info]   at scala.collection.mutable.ResizableArray.apply(ResizableArray.scala:43)
[info]   at scala.collection.mutable.ResizableArray.apply$(ResizableArray.scala:42)
[info]   at scala.collection.mutable.ArrayBuffer.apply(ArrayBuffer.scala:48)
[info]   at org.clulab.asist.StubActions.$anonfun$removeResearcher$1(AsistActions.scala:27)
[info]   at org.clulab.asist.StubActions.$anonfun$removeResearcher$1$adapted(AsistActions.scala:21)
[info]   at scala.collection.immutable.List.foreach(List.scala:389)
[info]   at org.clulab.asist.StubActions.removeResearcher(AsistActions.scala:21)
[info]   at sun.reflect.GeneratedMethodAccessor28.invoke(Unknown Source)
[info]   at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[info]   at java.lang.reflect.Method.invoke(Method.java:498)
[info]   ...
[info] - should Recognize fire entities *** FAILED ***
[info]   java.lang.IndexOutOfBoundsException: 0
[info]   at scala.collection.immutable.Vector.checkRangeConvert(Vector.scala:123)
[info]   at scala.collection.immutable.Vector.apply(Vector.scala:114)
[info]   at org.clulab.asist.TestParse.$anonfun$new$9(TestParse.scala:75)
[info]   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[info]   at org.scalatest.FlatSpecLike$$anon$1.apply(FlatSpecLike.scala:1682)
[info]   at org.scalatest.TestSuite.withFixture(TestSuite.scala:196)
[info]   ...
[info] - should Recognize infrastructure entities *** FAILED ***
[info]   java.lang.IndexOutOfBoundsException: 0
[info]   at scala.collection.immutable.Vector.checkRangeConvert(Vector.scala:123)
[info]   at scala.collection.immutable.Vector.apply(Vector.scala:114)
[info]   at org.clulab.asist.TestParse.$anonfun$new$10(TestParse.scala:81)
[info]   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[info]   at org.scalatest.FlatSpecLike$$anon$1.apply(FlatSpecLike.scala:1682)
[info]   at org.scalatest.TestSuite.withFixture(TestSuite.scala:196)
[info]   ...
[info] - should Recognize switch entities *** FAILED ***
[info]   java.lang.IndexOutOfBoundsException: 0
[info]   at scala.collection.immutable.Vector.checkRangeConvert(Vector.scala:123)
[info]   at scala.collection.immutable.Vector.apply(Vector.scala:114)
[info]   at org.clulab.asist.TestParse.$anonfun$new$11(TestParse.scala:90)
[info]   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[info]   at org.scalatest.FlatSpecLike$$anon$1.apply(FlatSpecLike.scala:1682)
[info]   at org.scalatest.TestSuite.withFixture(TestSuite.scala:196)
[info]   ...
[info] - should Recognize foe entities *** FAILED ***
[info]   java.lang.IndexOutOfBoundsException: 0
[info]   at scala.collection.immutable.Vector.checkRangeConvert(Vector.scala:123)
[info]   at scala.collection.immutable.Vector.apply(Vector.scala:114)
[info]   at org.clulab.asist.TestParse.$anonfun$new$12(TestParse.scala:98)
[info]   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[info]   at org.scalatest.FlatSpecLike$$anon$1.apply(FlatSpecLike.scala:1682)
[info]   at org.scalatest.TestSuite.withFixture(TestSuite.scala:196)
[info]   ...
[info] - should Recognize person entities *** FAILED ***
[info]   java.lang.reflect.InvocationTargetException:
[info]   at sun.reflect.GeneratedMethodAccessor28.invoke(Unknown Source)
[info]   at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[info]   at java.lang.reflect.Method.invoke(Method.java:498)
[info]   at scala.reflect.runtime.JavaMirrors$JavaMirror$JavaVanillaMethodMirror2.jinvokeraw(JavaMirrors.scala:384)
[info]   at scala.reflect.runtime.JavaMirrors$JavaMirror$JavaMethodMirror.jinvoke(JavaMirrors.scala:340)
[info]   at scala.reflect.runtime.JavaMirrors$JavaMirror$JavaVanillaMethodMirror.apply(JavaMirrors.scala:356)
[info]   at org.clulab.odin.impl.ActionMirror.$anonfun$reflect$1(ActionMirror.scala:23)
[info]   at org.clulab.odin.impl.GraphExtractor.findAllIn(Extractor.scala:84)
[info]   at org.clulab.odin.impl.Extractor.$anonfun$findAllIn$1(Extractor.scala:19)
[info]   at org.clulab.odin.impl.Extractor.$anonfun$findAllIn$1$adapted(Extractor.scala:18)
[info]   ...
[info]   Cause: java.lang.IndexOutOfBoundsException: 0
[info]   at scala.collection.mutable.ResizableArray.apply(ResizableArray.scala:43)
[info]   at scala.collection.mutable.ResizableArray.apply$(ResizableArray.scala:42)
[info]   at scala.collection.mutable.ArrayBuffer.apply(ArrayBuffer.scala:48)
[info]   at org.clulab.asist.StubActions.$anonfun$removeResearcher$1(AsistActions.scala:27)
[info]   at org.clulab.asist.StubActions.$anonfun$removeResearcher$1$adapted(AsistActions.scala:21)
[info]   at scala.collection.immutable.List.foreach(List.scala:389)
[info]   at org.clulab.asist.StubActions.removeResearcher(AsistActions.scala:21)
[info]   at sun.reflect.GeneratedMethodAccessor28.invoke(Unknown Source)
[info]   at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[info]   at java.lang.reflect.Method.invoke(Method.java:498)
[info]   ...
[info] Run completed in 28 seconds, 205 milliseconds.
[info] Total number of tests run: 13
[info] Suites: completed 1, aborted 0
[info] Tests: succeeded 0, failed 13, canceled 0, ignored 0, pending 0
[info] *** 13 TESTS FAILED ***
[error] Failed tests:
[error] 	org.clulab.asist.TestParse
[error] (Test / test) sbt.TestsFailedException: Tests unsuccessful
[error] Total time: 64 s, completed Oct 22, 2020 10:37:51 AM
[INFO] [10/22/2020 10:37:51.914] [Thread-2] [CoordinatedShutdown(akka://sbt-web)] Starting coordinated shutdown from JVM shutdown hook
```

Are you able to reproduce this on your computer?

@jastier Do you get these errors as well? They might hamper your setting up of continuous integration.

**Comment by jastier** [comment]
I'm working this issue, will advise.

**Comment by pelovett** [comment]
After discussing these errors with @jastier over email, they seem to be caused by two issues. The first issue is that the test logic doesn't check the size of arrays before indexing them, which causes indexing errors which make the test results unreadable. The second issue we found is that the app raises errors when the user tries to use a Java version other than Java 1.8. 

The first issue can trivially be solved by adding a length check to the tests. The second issue will need to be solved either by fixing the dependency issue or by asserting that the code is run with Java 1.8.

**Comment by adarshp** [comment]
Closed by #4 ,
